psi4/src/psi4/libfock/jk.cc

```diff
@@ -145,6 +155,9 @@ std::shared_ptr<JK> JK::build_JK(std::shared_ptr<BasisSet> primary, std::shared_
         if (options["DF_INTS_NUM_THREADS"].has_changed())
             jk->set_df_ints_num_threads(options.get_int("DF_INTS_NUM_THREADS"));
 
+        // QUESTION: Should Direct JK use incfock by default or not?
+        // if (!options["INCFOCK"].has_changed()) jk->set_incfock(true);
```

Comment:
I noticed your question here, and I think there are two points to be made about this:

1. For a lot of more simplistic IncFock implementations, IncFock acts as a double-edged sword - while basic IncFock can greatly speed up SCF iterations close to convergence in conjunction with density screening, it can also notably increase the number of SCF iterations needed to converge in the first place. However, the IncFock implementation in Psi4 has some nice bells and whistles - particularly, using the full matrices every couple of iterations and, more importantly, disabling IncFock entirely when a certain convergence point has been reached. These bells and whistles make Psi4's IncFock much more stable and reliable; thus, IncFock isn't nearly the same double-edged sword that it often can be.
2. IncFock is especially powerful in conjunction with Density screening, since Density screening becomes much more prevelant in later SCF iterations when the difference density matrix, which approaches the zero matrix as the SCF procedure reaches convergence, is used. Thus, in conjunction with the first point, it could make the most sense to tie the default status of IncFock to the type of screening used - true if density screening is used, and false otherwise. This obviously raises the question of what to do with JK algos that have density screening hard-baked into them, such as LinK. In those cases, if we are taking the approach outlined in this comment, it would likely make the most sense and be most consistent if we enabled IncFock by default for those methods.

Those are my thoughts on the matter.

Reply:
For now, I will enable `incfock` for DirectJK only if there is density screening, and false otherwise. For DFJCOSK, `incfock` will be `true` by default.